src/test_typing_extensions.py

```diff
@@ -7246,6 +7292,171 @@ def test_getitem(self):
         self.assertEqual(get_args(fully_subscripted), (Iterable[float],))
         self.assertIs(get_origin(fully_subscripted), ListOrSetT)
 
+        # Test ParamSpec and Ellipsis
+        P = ParamSpec('P')
+        CallableP = TypeAliasType("CallableP", Callable[P, Any], type_params=(P,))
+        # () -> Any
+        callable_no_arg = CallableP[[]]
+        self.assertEqual(get_args(callable_no_arg), ([],))
+        # (int) -> Any
+        callable_arg = CallableP[int]
+        self.assertEqual(get_args(callable_arg), (int,))
+
+        callable_arg_list = CallableP[[int]]
+        self.assertEqual(get_args(callable_arg_list), ([int],))
+
+        # (int, int) -> Any
+        callable_arg2 = CallableP[int, int]
+        self.assertEqual(get_args(callable_arg2), (int, int,))
+
+        callable_arg2_list = CallableP[[int, int]]
+        self.assertEqual(get_args(callable_arg2_list), ([int, int],))
+        # (...) -> Any
+        callable_ellipsis = CallableP[...]
+        self.assertEqual(get_args(callable_ellipsis), (...,))
+
+        callable_ellipsis2 = CallableP[(...,)]
+        self.assertEqual(callable_ellipsis, callable_ellipsis2)
+        # (int, ...) -> Any
+        callable_arg_more = CallableP[[int, ...]]
+        self.assertEqual(get_args(callable_arg_more), ([int, ...],))
```

Comment:
This is really the same test being done multiple times, but with different inputs to the test. This can be made more readable by using `subTest`, which has the added advantage that one specific test case can fail without failing all the other test cases at the same time: it will report a specific failure for one subtest, but continue to run the test for all the other cases:

```suggestion
        get_args_test_cases = {
            # () -> Any
            CallableP[[]]: ([],),
            # (int) -> Any
            CallableP[int]: (int,),
            CallableP[[int]]: ([int],),
            # (int, int) -> Any
            CallableP[int, int]:  ([int],),
            CallableP[[int, int]]: ([int, int],),
            # (...) -> Any
            CallableP[...]: (...,),
            # (int, ...) -> Any
            CallableP[[int, ...]]: ([int, ...],)
        }
        
        for index, (expression, expected_args) in enumerate(get_args_test_cases.items()) {
            with self.subTest(index=index, expression=expression):
                self.assertEqual(get_args(expression), expected_args))

        self.assertEqual(CallableP[...], CallableP[(...,)])
```

Reply:
Thank you great idea. I was not familiar with this yet. Added the subtests. However, if the argument is a list, e.g. CallableP[[int, ...]], the alias is not hashable and a dict cannot be used here. A list of lists looked worse to me.